cli/cmd/lql.go

```diff
@@ -198,6 +199,14 @@ func inputQuery(cmd *cobra.Command) (string, error) {
 	if queryCmdState.URL != "" {
 		return inputQueryFromURL(queryCmdState.URL)
 	}
+	// if running via stdin
+	stat, err := os.Stdin.Stat()
+	if err != nil {
+		cli.Log.Debugw("error retrieving stdin mode")
+	} else if (stat.Mode() & os.ModeCharDevice) == 0 {
```

Comment:
I have never used `os.ModeCharDevice` but instead this:
```suggestion
	} else if (stat.Mode() & os.ModeNamedPipe) != 0 {
```

From the official Golang documentation https://pkg.go.dev/os
```
	ModeDevice     = fs.ModeDevice     // D: device file
	ModeNamedPipe  = fs.ModeNamedPipe  // p: named pipe (FIFO)
	ModeCharDevice = fs.ModeCharDevice // c: Unix character device, when ModeDevice is set
```

I have used `ModeDevice` and `ModeCharDevice` for files, not pipes. But if it works, Im +1

Reply:
I will try this again; thanks!

Reply:
So `os.ModeNamedPipe` is having a problem with the test setup as well.  I think I'm going to stick with the slightly more robust `else if (stat.Mode() & os.ModeCharDevice) == 0`...